src/types/global.ts

```diff
@@ -16,3 +16,36 @@ export interface ParaTaskCard {
 
 export type Goal = SelectableForTable<"goal">;
 export type Subgoal = SelectableForTable<"subgoal">;
+export type Student = SelectableForTable<"student">;
+export type Para = SelectableForTable<"user">;
+export type Iep = SelectableForTable<"iep">;
+export type StudentWithIep = Student & Iep;
+
+export interface HeadCell {
+  id: string;
+  label: string;
+  hasInput: boolean;
+}
+
+export interface StudentHeadCell extends HeadCell {
+  id: keyof Student;
+}
+
+export interface ParaHeadCell extends HeadCell {
+  id: keyof Para;
+}
+
+export interface IepHeadCell extends HeadCell {
+  id: keyof Iep;
+}
+
+export type StudentWithIepHeadcell = StudentHeadCell | IepHeadCell;
```

Comment:
🐧 I would probably keep these in `table/table.tsx` since they're specific to that component

Reply:
Some of these types are being used in the `student/index` and `staff/index` pages. Would you keep the base `HeadCell` in this file, and move types that are only used in one file to that file?

Reply:
it's generally best to keep the type close to the thing that "owns" it--by ownership I mean the thing that takes parameters of the given type and results in terminal behavior

in this case the table component owns the cell types

components that want to render a table can import the cell types from the table component

another way to look at it is what types you would have to change if you want to refactor a particular component